cmd/shutdown.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+const (
+	shutdownTimeout = 10 * time.Second
+)
+
+//GracefulShutdown implements releasing all resouces it got from system, finish all request handling and return responses when service stopping.
+func gracefulShutdown(done chan<- bool) error {
```

Comment:
Do you really need `done` channel if you close it inside function and waiting outside?

Reply:
Chan used for blocking main until graceful shutdown will close all connections

Reply:
So you can call `gracefulShutdown()` and wait

Reply:
yep